gazebo_plugins/src/gazebo_ros_force.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/common/Events.hh>
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/Model.hh>
+#include <gazebo_plugins/gazebo_ros_force.hpp>
+#include <gazebo_ros/conversions.hpp>
+#include <gazebo_ros/node.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <memory>
+#include <string>
+
+namespace gazebo_plugins
+{
+class GazeboRosForcePrivate
+{
+public:
+  /// A pointer to the Link, where force is applied
+  gazebo::physics::LinkPtr link_;
+
+  /// A pointer to the GazeboROS node.
+  gazebo_ros::Node::SharedPtr ros_node_;
+
+  /// Wrench subscriber
+  rclcpp::Subscription<geometry_msgs::msg::Wrench>::SharedPtr wrench_sub_;
+
+  /// Container for the wrench force that this plugin exerts on the body.
+  geometry_msgs::msg::Wrench wrench_msg_;
+
+  // Pointer to the update event connection
+  gazebo::event::ConnectionPtr update_connection_;
+};
+
+GazeboRosForce::GazeboRosForce()
+: impl_(std::make_unique<GazeboRosForcePrivate>())
+{
+  impl_->wrench_msg_.force =
+    gazebo_ros::Convert<geometry_msgs::msg::Vector3>(ignition::math::Vector3d::Zero);
+  impl_->wrench_msg_.torque =
+    gazebo_ros::Convert<geometry_msgs::msg::Vector3>(ignition::math::Vector3d::Zero);
```

Comment:
I believe these will already be zeroed out by the default constructor

Reply:
Oh you're right, [here](https://design.ros2.org/articles/generated_interfaces_cpp.html)'s a handy reference. I'll update.

Reply:
https://github.com/ros-simulation/gazebo_ros_pkgs/pull/784/commits/0cc0e06f4aaca8b9cabd025dde907402afc9fc7a